No install target in cmake scripts

**Bug Description:**
This was one of TODO items in https://github.com/surge-synthesizer/surge/issues/1206
However, it was closed without this itemn being fixed.

**Surge Version**
1.9.0

The FreeBSD port currently has this script to install Surge:
```
do-install:
        # LV2 plugin
        ${MKDIR} ${STAGEDIR}${PREFIX}/lib/lv2
        cd ${BUILD_WRKSRC}/surge_products && ${COPYTREE_SHARE} Surge.lv2 ${STAGEDIR}${PREFIX}/lib/lv2
        ${STRIP_CMD} ${STAGEDIR}${PREFIX}/lib/lv2/Surge.lv2/Surge.so
        # VST plugin
        ${MKDIR} ${STAGEDIR}${PREFIX}/lib/vst
        ${INSTALL_LIB} ${BUILD_WRKSRC}/surge_products/Surge.vst3/Contents/*/Surge.so ${STAGEDIR}${PREFIX}/lib/vst
        # data
        cd ${WRKSRC}/resources/data && ${COPYTREE_SHARE} . ${STAGEDIR}${DATADIR}
```

This is error-prone because something can be missed in it.

```build-linux.sh``` doesn't work for the port because it assumes fixed paths instead of paths that cmake uses.

Now I tried ```install-vst3-global``` but it breaks:
```
[ 16% 2/6] cd /disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0 && echo Installing\ local\ resources\ into\ /usr/local/share/surge && /usr/local/bin/cmake -E make_directory /usr/local/share/surge && rsync -r --delete resources/data/ /usr/local/share/surge/
FAILED: CMakeFiles/install-resources-global.util 
cd /disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0 && echo Installing\ local\ resources\ into\ /usr/local/share/surge && /usr/local/bin/cmake -E make_directory /usr/local/share/surge && rsync -r --delete resources/data/ /usr/local/share/surge/
Installing local resources into /usr/local/share/surge
Error creating directory "/usr/local/share/surge".
[ 33% 2/6] cd /disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0 && /usr/local/bin/cmake -D PROJECT_VERSION_MAJOR=1 -D PROJECT_VERSION_MINOR=9 -D SURGESRC=/disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0 -D SURGEBLD=/disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/.build -D AZURE_PIPELINE= -D WIN32= -D CMAKE_INSTALL_PREFIX=/usr/local -D SURGE_COMPILER_INFO="Clang-10.0.1" -P /disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0/cmake/versiontools.cmake
-- VERSION_GIT_INFO file is present; using that rather than git query
-- Setting up surge version
--   git hash is 0000000 and branch is unknown
--   buildhost is yv.noip.me
--   buildarch is amd64
-- Developer Local Build
-- Using SURGE_VERSION=1.9.unknown.0000000
-- Configuring /disk-samsung/freebsd-ports/audio/surge-synthesizer-lv2/work/.build/geninclude/version.cpp
ninja: build stopped: subcommand failed.
```

It doesn't use proper stage directory and installs into the read-only ```$PREFIX```.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,10 +136,10 @@ endif()
 
 if( SURGE_ALTERNATE_JUCE )
   message( STATUS "Using *Alternate* JUCE from ${SURGE_ALTERNATE_JUCE}" )
-  add_subdirectory( ${SURGE_ALTERNATE_JUCE} ${CMAKE_BINARY_DIR}/alternate-juce )
+  add_subdirectory( ${SURGE_ALTERNATE_JUCE} ${CMAKE_BINARY_DIR}/alternate-juce  EXCLUDE_FROM_ALL)
 else()
   message( STATUS "Using JUCE from submodule libs/JUCE" )
-  add_subdirectory( libs/JUCE )
+  add_subdirectory( libs/JUCE EXCLUDE_FROM_ALL )
 endif()
 
 include(cmake/stage-extra-content.cmake)
@@ -1075,6 +1075,30 @@ if( WIN32 )
   endif()
 endif()
 
+# Add CMake Install rules for LINUX only right now (mac and win use an installer)
+if( UNIX AND NOT APPLE)
+  if( BUILD_SURGE_XT)
+    install(TARGETS surge-xt_Standalone DESTINATION bin)
+    # Alas you would think you could just say 
+    # install(TARGETS surge-xt_VST3 DESTINATION lib/vst3 )
+    # but you can't because surge-xt_VST3 points to the .so inside the VST3 bundle so instead say
+    install( DIRECTORY "${SURGE_XT_OUTPUT_DIR}/VST3/Surge XT.vst3" DESTINATION lib/vst3 )
+    if( JUCE_SUPPORTS_LV2 )
+      install(TARGETS surge-xt_LV2 DESTINATION lib/lv2 )
+    endif()
+  endif()
+  if( BUILD_SURGE_EFFECTS_BANK )
+    install(TARGETS surge-fx_Standalone DESTINATION bin)
+    # and again
+    # install(TARGETS surge-fx_VST3 DESTINATION lib/vst3 )
+    install( DIRECTORY "${SURGE_FX_OUTPUT_DIR}/VST3/Surge XT Effects.vst3" DESTINATION lib/vst3 )
+   
+    if( JUCE_SUPPORTS_LV2 )
+      install(TARGETS surge-fx_LV2 DESTINATION lib/lv2)
+    endif()
+  endif()
+  install(DIRECTORY ${CMAKE_SOURCE_DIR}/resources/data/ DESTINATION share/surge-xt)
+endif()
 
 
 if( ${BUILD_SURGE_PYTHON_BINDINGS} )

--- a/README.md
+++ b/README.md
@@ -183,6 +183,19 @@ This is off by default with CMake JUCE but you can turn it on with `-DSURGE_COPY
 (`~/.vst3` on linux, '~/Library/Audio/Plugins` on mac). On windows it will attempt to install the VST3
 so setting this option may require admin privileges in your build environment.
 
+### CMake Install Targets (Linux and other non-apple unixes only)
+
+On systems which are `UNIX AND NOT APPLE`, the cmake file provides an install target which will install
+all needed assets to the `CMAKE_INSTALL_PREFIX`. This means a complete install can be accomplished by
+
+```
+cmake -Bignore/sxt -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr
+cmake --build ignore/sxt --config Release --parallel 8
+sudo cmake --install ignore/sxt
+```
+
+and you should get a working install in `/usr/bin`, `/usr/share` and `/usr/lib`
+
 ### Installing assets (unixes only)
 
 The targets `install-resources-local` and `install-resources-global` install the plugin resources

